test/fs_test.go

```diff
@@ -0,0 +1,397 @@
+package test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path"
+	"testing"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+)
+
+func TestStateWrapper_OpenInvalidPath(t *testing.T) {
+	st := llb.Scratch().File(llb.Mkfile("/bar", 0644, []byte("hello world")))
+	testEnv.RunTest(context.Background(), t, func(ctx context.Context, gwc gwclient.Client) (*gwclient.Result, error) {
```

Comment:
I may have messed up some tests, but the should use `baseCtx` rather than` context.Background()`